feat: add GeoDNS documentation (#842)
coder/docs: admin/satellites/geo-dns.md

---

> title: GeoDNS

Reviewer: How about we call this "global load balancing," or something similar?

If you google GeoDNS, this is the top hit: https://en.wikipedia.org/wiki/GeoDNS

Do these docs only refer to Anycast routing, or can people use this if they are using split-horizon DNS?

description: |
    Learn how to configure your primary and satellite deployments to share a
    single hostname.
state: alpha
---

Out of the box, satellites deployments have a different access URL to the

> primary deployment. This may cause confusion for engineering teams located

Reviewer: This says "This may cause confusion" twice

globally. This may cause confusion amongst users wondering which access URL they
should access Coder from.

Coder supports a GeoDNS configuration where all satellites and the primary

> deployment share the same hostname, so all users access Coder using a consistent

Reviewer: needs a passage about users being routed to the nearest installation or satellite, ensuring low latency

URL but maintain low latency to their workspaces.

> We recommend maintaining separate access URLs for each satellite and primary
> deployment alongside the new GeoDNS access URL so you can access specific
> deployments while avoiding GeoDNS in the future. This guide will walk you
> through this configuration process.

## Requirements

You will need the following:

- A primary access URL (e.g. `https://primary.example.com`).
- One or more satellite access URLs (e.g. `https://sydney.example.com`,
  `https://london.example.com`).

> - A GeoDNS load balancer access URL (e.g. `https://coder.example.com`). The

Reviewer: GCP does something similar with Anycast, https://cloud.google.com/load-balancing

We should distinguish whether this feature is for people using Anycast (a CDN or global load balancing) versus split-horizon DNS (where the DNS responses vary according to the user's IP address/location)

  default backend should be set to the primary access URL or IP address. The
  backend for each region with a satellite should be set to the satellite access
  URL or IP address.
- The TLS certificate used by the primary must have both the primary hostname
  and the GeoDNS hostname
- The TLS certificate used by each satellite must have the satellite's hostname
  and the GeoDNS hostname

> If you are using cert-manager, you can easily add more hostnames to your
> certificate by adding them to the `spec.dnsNames` section.

To create a GeoDNS load balancer on Cloudflare, you can refer to our simple
instructions [below](#create-a-geodns-load-balancer-on-cloudflare).

## Configure GeoDNS on Coder

> To configure GeoDNS onn Coder:

Reviewer: ```suggestion
To configure GeoDNS on Coder:
```

> 1. In the primary helm values file set `coderd.alternateHostnames` to your

Reviewer: ```suggestion
1. In the primary helm values file, set `coderd.alternateHostnames` to your
```

   primary hostname and your GeoDNS hostname:

   ```yaml
   coderd:
     alternateHostnames:
       - "primary.example.com"
       - "coder.example.com"
   ```

1. In each satellite helm values file:

>     1. Set `coderd.satellite.accessURL` to your GeoDNS URL (this value is used

Reviewer: ```suggestion
    1. Set `coderd.satellite.accessURL` to your primary access URL, which is the name of the global load balancer in front of Coder deployments (this value is used
```

       as the default URL)
    1. Set `coderd.alternateHostnames` to your satellite's specific hostname and
       your GeoDNS hostname:

       ```yaml
       coderd:
         alternateHostnames:
           - "satellite.example.com"
           - "coder.example.com"
       ```

> 1. Deploy your primary and satellites with your new helm values.

Reviewer: ```suggestion
1. Deploy your primary and satellites with your new Helm values.
```
